src/ui_iteminfo.cpp

```diff
@@ -0,0 +1,153 @@
+
+    if( !data.get_item_name().empty() ) {
+        window.draw_colored_text( data.get_item_name(), base_color, wrap_width );
+        ImGui::Spacing();
+    }
+    if( !data.get_type_name().empty() &&
+        data.get_item_name().find( data.get_type_name() ) == std::string::npos ) {
+        window.draw_colored_text( data.get_type_name(), base_color, wrap_width );
+    }
+
+    bool same_line = false;
+    for( const iteminfo &i : data.get_item_display() ) {
+        if( i.sType == "DESCRIPTION" ) {
+            if( i.bDrawName ) {
+                if( i.sName == "--" ) {
+                    ImGui::Separator();
+                } else {
+                    window.draw_colored_text( i.sName, base_color, wrap_width );
+                }
+            }
+        } else {
+            if( same_line ) {
+                ImGui::SameLine( 0, 0 );
+            }
+            std::string formatted_string;
+            if( i.bDrawName ) {
+                formatted_string = i.sName;
+            }
+
+            std::string sFmt = i.sFmt;
+
+            //A bit tricky, find %d and split the string
+            size_t pos = sFmt.find( "<num>" );
+            if( pos != std::string::npos ) {
+                formatted_string += sFmt.substr( 0, pos );
+            } else {
+                formatted_string += sFmt;
+            }
+
+            if( i.sValue != "-999" ) {
+                nc_color this_color = c_yellow;
+                for( const iteminfo &k : data.get_item_compare() ) {
+                    if( k.sValue != "-999" ) {
+                        if( i.sName == k.sName && i.sType == k.sType ) {
+                            double iVal = i.dValue;
+                            double kVal = k.dValue;
+                            if( i.sFmt != k.sFmt ) {
+                                // Different units, compare unit adjusted vals
+                                iVal = i.dUnitAdjustedVal;
+                                kVal = k.dUnitAdjustedVal;
+                            }
+                            if( iVal > kVal - .01 &&
+                                iVal < kVal + .01 ) {
+                                this_color = c_light_gray;
+                            } else if( iVal > kVal ) {
+                                if( i.bLowerIsBetter ) {
+                                    this_color = c_light_red;
+                                } else {
+                                    this_color = c_light_green;
+                                }
+                            } else if( iVal < kVal ) {
+                                if( i.bLowerIsBetter ) {
+                                    this_color = c_light_green;
+                                } else {
+                                    this_color = c_light_red;
+                                }
+                            }
+                            break;
+                        }
+                    }
+                }
```

Comment:
Can we split this out into a separate function? Something that takes the item and compare items by reference and returns a color? That would make the whole function a lot easier to read.

Reply:
I simply reused what was already given in `draw_item_info` and I'm not wiling to rewrite how iteminfo works internally at all. But that's definitely a separate PR.